Employee component -> New Employee -> Edit Employee

zerokrezan opened the pull request:
Employee component -> New Employee -> Edit Employee

zerokrezan committed:
update employee component page

zerokrezan committed:
update employee component page

zerokrezan committed:
update employee component page

zerokrezan committed:
Merge branch 'main' into Employee-Component

# Conflicts:
#	src/app/app.component.html
#	src/app/app.component.ts

zerokrezan committed:
update employee component page

zerokrezan committed:
update employee component page

pichaneeschmidt commented on src/app/employee/employee.component.ts:
```diff
@@ -1,24 +1,46 @@
   employee!: Employee;
+  configuration!: TableConfiguration<Employee>;
+  qualifications: WritableSignal<Qualification[]> = signal(
+    [new Qualification(1, 'Java'), new Qualification(2, 'Python'), new Qualification(3, 'C++'),
```
Why don't you get the data from the employee?
When I understand correctly....
The qualifications here should belong to the employee, right?

zerokrezan replied:
yes you are absolutely right, that was only for testing, will fix it then :)

zerokrezan replied:
already solved!

Commit:
bugfix caching services and example for data display in form

zerokrezan committed:
updated navbar:
changed employees to employee/all to activate employee a tag for all pages which include employee within their url by adding this:
[routerLinkActiveOptions]="{exact:false}"

zerokrezan committed:
updated Employee Component -> use qualifications of current employee

ZneroI commented on src/app/app.component.ts:
```diff
@@ -3,16 +3,19 @@ import { FormsModule } from '@angular/forms';
 import { CommonModule } from '@angular/common';
 import {NavbarComponent} from "./components/navbar/navbar.component";
 import { RouterLink, RouterModule, RouterOutlet } from '@angular/router';
+import {EmployeeComponent} from "./employee/employee.component";
```
this file is by accident part of the commit. delete unnecessary imports

zerokrezan replied:
Done

ZneroI commented on src/app/app.routes.ts:
```diff
@@ -2,10 +2,12 @@ import {Routes} from '@angular/router';
-
-export const routes: Routes = [ 
-  {path: 'employees', title: "employees", component: EmployeeListComponent, canActivate: [AppAuthGuard]}, 
+export const routes: Routes = [
```
please rework the title properties. its "employees" everywhere.

zerokrezan replied:
Done

ZneroI commented on src/app/components/navbar/navbar.component.html:
```diff
@@ -3,12 +3,12 @@
     <div class="collapse navbar-collapse">
```
be careful when merging. props just take the master branch version, except maybe for the values of routerLink

zerokrezan replied:
took the main branch version

ZneroI commented on src/app/components/navbar/navbar.component.ts:
```diff
@@ -1,5 +1,5 @@
-import {Component} from '@angular/core';
```
this file should not be part of the pull request. take master branch version

zerokrezan replied:
took the main branch version

ZneroI commented on src/app/employee/employee.component.html:
```diff
@@ -1,13 +1,64 @@
-<div>
```
a simple way to get the same marign on both sides of the screen would be a page-wrapper element. maybe have a look in employee-list in master

zerokrezan replied:
TODO for later

ZneroI commented on src/app/employee/employee.component.html:
```diff
@@ -1,13 +1,64 @@
-<div>
-  <h1>Employee Form Page</h1>
+<div class="container-fluid">
```
there is a global css style that gives left and right margin of 5vw (in master that is)

ZneroI commented on src/app/employee/employee.component.ts:
```diff
@@ -1,39 +1,105 @@
+      console.log("Employee loaded:", this.emp);
+    });
+
+    //this.employee = new Employee();
```
delete commented out code

zerokrezan replied:
done

ZneroI commented on src/app/employee/employee.component.ts:
```diff
@@ -1,39 +1,105 @@
   ngOnInit(): void {
-    this.employee = new Employee();
+    this.id = this.activatedRoute.snapshot.params['id'];
+    this.employeeService.select(this.id).subscribe((employee: Employee) => {
```
please add OnDestroy and unsubscribe there. save subscription in instance-variable

zerokrezan replied:
DONE

ZneroI commented on src/app/employee/employee.component.ts:
```diff
@@ -1,39 +1,105 @@
+  formDataEmployee: WritableSignal<Employee> = signal(new Employee());
 
+  emp!: Employee;
+  employee!: Signal<Employee|undefined>;
```
take a look which of the variables you are actually using. delete the rest. for instance the employee-signal

zerokrezan replied:
the second employee is needed when using EmployeeCacheService

zerokrezan replied:
removed!

ZneroI commented on src/app/employee/employee.component.ts:
```diff
@@ -1,39 +1,105 @@
+    // }
     this.employeeCacheService.refresh();
+
+    this.employee = computed(()=> {
```
if this doesnt work or doesnt get used, delete

zerokrezan replied:
removed!

ZneroI commented on src/app/services/qualification.service.ts:
```diff
@@ -15,7 +15,7 @@ export class QualificationService {
   select(id: number): Observable<Qualification> {
     let q: Qualification = new Qualification();
```
uuuh i think i made a mistake there when i wrote the service. "q" is not used🙄. but its not critical

zerokrezan committed:
updated Employee Component
